src/server/hll_family.cc

```diff
@@ -211,13 +216,78 @@ void PFCount(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+OpResult<int> PFMergeInternal(CmdArgList args, ConnectionContext* cntx) {
+  try {
```

Comment:
I do not think you should catch OOM exceptions in coordinator thread.
I believe such catch belongs to the logic that does on shard execution.

Reply:
Ah, I see now. You are worried that all the temporary values will OOM. In that case I would still reduce the scope of `try` to be around first execute hop, which allocates the most. please note that you still need to proceed with the second hop anyways - otherwise we will reach a deadlock.

Reply:
You're right, I actually think that the best place to guard is inside the callback, inside `ReadValues()`.
This is the major source of allocation, and it also has the nice side effect of also guarding `PFCOUNT`.